src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,118 @@
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
 export const TodoModal: React.FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setLoading] = useState(false);
+  const [didload, setLoaded] = useState(false);
+  const [hasError, setError] = useState(false);
+  const dispatch = useAppDispatch();
+  const {
+    userId,
+    completed,
+    id,
+    title,
+  } = useAppSelector(state => state.currentTodo) as Todo;
+
+  useEffect(() => {
+    const loadUser = async () => {
+      setLoading(true);
+      try {
+        const userFromServer = await getUser(userId);
+
+        setUser(userFromServer);
+        setLoaded(true);
+      } catch (err) {
+        setError(true);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    loadUser();
+  }, [id]);
+
+  const showTodoInfo = !isLoading && !hasError && didload;
+
+  const closeModal = () => {
+    dispatch(actions.removeTodo());
+  };
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      <Loader />
+      {isLoading && <Loader />}
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+      {showTodoInfo && (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              {`Todo #${id}`}
+            </div>
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={closeModal}
+            />
+          </header>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">fugiat veniam minus</p>
+          <div className="modal-card-body">
+            <p
+              className="block"
+              data-cy="modal-title"
+            >
+              {title}
+            </p>
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+            <p className="block" data-cy="modal-user">
+              {completed
+                ? (
+                  <strong className="has-text-success">Done</strong>
+                ) : (
+                  <strong className="has-text-danger">Planned</strong>)}
+              {' by '}
+              <a
+                href={`mailto:${user?.email}`}
+              >
```

Comment:
```suggestion
              <a href={`mailto:${user?.email}`}>
```

Reply:
@mgrinko Thank you for the suggestions. I applied them.